pyproject.toml

```
@@ -86,13 +87,13 @@ write_to_template = '__version__ = "{version}"'
 
 [tool.black]
 line-length = 100
-target-version = ['py39']
+target-version = ['py312']
```

we should probably target the minimum supported version instead

```suggestion
target-version = ['py310']
```